servers/src/main/java/tachyon/worker/block/evictor/LFUEvictor.java

```diff
@@ -0,0 +1,231 @@
+   * with this node, append it to the end of next layer, otherwise, create a new next layer with the
+   * Node as the only element
+   *
+   * If the access count arrives at limit of {@code long}, it will no longer be increased
+   */
+  class CountNode extends DoubleLinkListNode {
+    long mCount;
+    Node mFirst;
+    Node mLast;
+
+    CountNode() {
+      super();
+      mCount = 0;
+      mFirst = mLast = null;
+    }
+
+    CountNode(long count) {
+      super();
+      mCount = count;
+      mFirst = mLast = null;
+    }
+
+    void appendNode(Node node) {
+      if (mFirst == null) {
+        mFirst = node;
+      } else {
+        mLast.append(node);
+      }
+      mLast = node;
+      node.mHead = this;
+    }
+
+    CountNode nextCountNode() {
+      return (CountNode) mNext;
+    }
+  }
+
+  class Node extends DoubleLinkListNode {
+    long mBlockId;
+    CountNode mHead;
+
+    Node() {
+      super();
+      mBlockId = -1;
+      mHead = null;
+    }
+
+    Node(long id) {
+      super();
+      mBlockId = id;
+      mHead = null;
+    }
+
+    /** assume next layer exists and have elements other than the count node */
+    void appendToNextLayer() {
+      remove();
+      nextLayerHead().appendNode(this);
+    }
+
+    long count() {
+      return mHead.mCount;
+    }
+
+    Node first() {
+      return mHead.mFirst;
+    }
+
+    CountNode nextLayerHead() {
+      return mHead.nextCountNode();
+    }
+
+    Node nextNode() {
+      return (Node) mNext;
+    }
+  }
+
+  /** Always be the head of the first layer */
+  private CountNode mHead;
+  /** Map from blockId to corresponding Node */
+  private Map<Long, Node> mCache;
+
+  public LFUEvictor(BlockMetadataManager meta) {
+    mMeta = meta;
+
+    mHead = new CountNode();
+    mCache = new HashMap<Long, Node>();
+  }
+
+  /** Complexity: O(1) */
+  @Override
+  public void onAccessBlock(long userId, long blockId) {
+    synchronized (mLock) {
+      if (mCache.containsKey(blockId)) {
+        Node node = mCache.get(blockId);
+        if (node.count() == Long.MAX_VALUE) {
+          return;
+        }
+        if (node.nextLayerHead() == null && node.first() == node) { // only Node in last layer
+          node.mHead.mCount += 1;
+        } else {
+          // not the only Node in last layer
+          // or count of next layer is not its current count plus 1
+          if (node.nextLayerHead() == null || node.nextLayerHead().mCount != node.count() + 1) {
+            // create new layer
+            node.mHead.append(new CountNode(node.count() + 1));
+          }
+          node.appendToNextLayer();
+        }
+      } else {
+        Node node = new Node(blockId);
+        mHead.appendNode(node);
+        mCache.put(blockId, node);
+      }
+    }
+  }
+
+  /**
+   * Evict layer by layer, in each layer, evict from first to last
+   */
+  @Override
+  public EvictionPlan freeSpace(long availableBytes, BlockStoreLocation location) throws IOException {
```

Comment:
I think the most intuitive parameter for `freeSpace` is `bytesToFree` instead of current `availableBytes`.  

Also for https://github.com/amplab/tachyon/blob/worker_wip/servers/src/main/java/tachyon/worker/block/TieredBlockStore.java#L205, maybe `assureAvailableSpace` is a more appropriate name.